common/src/test/java/com/linecorp/centraldogma/common/PathPatternBuilderTest.java

```diff
@@ -0,0 +1,87 @@
+        assertThatNoException().isThrownBy(() -> PathPattern.builder().hasExtension(".JPG"));
+        assertThatNoException().isThrownBy(() -> PathPattern.builder().hasExtension(".7z"));
+
+        assertThatThrownBy(() -> PathPattern.builder().hasExtension("가txt"))
+                .isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> PathPattern.builder().hasExtension("..tx.t"))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testSingleOption() {
+        assertThat(PathPattern.builder().startsWith("/foo/bar").build()
+                              .patternString()).isEqualTo("/foo/bar/**");
+        assertThat(PathPattern.builder().endsWith("foo.json").build()
+                              .patternString()).isEqualTo("/**/foo.json");
+        assertThat(PathPattern.builder().contains("/bar").build()
+                              .patternString()).isEqualTo("/**/bar/**");
+        assertThat(PathPattern.builder().hasExtension("json").build()
+                              .patternString()).isEqualTo("/**/*.json");
+    }
+
+    @Test
+    void testPathPatternBuilder() {
+        assertThat(PathPattern.builder()
+                              .startsWith("/foo/bar")
+                              .endsWith("foo.txt")
+                              .build()
+                              .patternString()).isEqualTo("/foo/bar/**/foo.txt");
+        assertThat(PathPattern.builder()
+                              .startsWith("/foo")
+                              .contains("/bar/")
+                              .hasExtension("json")
+                              .build()
+                              .patternString()).isEqualTo("/foo/**/bar/**/*.json");
+
+        assertThat(PathPattern.builder()
+                              .startsWith("/foo")
+                              .endsWith("qux.json")
+                              .hasExtension("json")
+                              .build()
+                              .patternString()).isEqualTo("/foo/**/*.json");
+        assertThat(PathPattern.builder()
+                              .startsWith("/foo")
+                              .hasExtension("json")
+                              .endsWith("qux.json")
+                              .build()
+                              .patternString()).isEqualTo("/foo/**/qux.json");
+
+        assertThat(PathPattern.builder()
+                              .contains("/foo")
+                              .contains("/bar")
+                              .build()
+                              .patternString()).isEqualTo("/**/foo/**/bar/**");
+        assertThat(PathPattern.builder()
+                              .startsWith("/foo/bar")
+                              .startsWith("/override")
+                              .hasExtension("json")
+                              .build()
+                              .patternString()).isEqualTo("/override/**/*.json");
```

Comment:
I think this can be quite confusing to users. Should we always override, even for `contains()`?

Reply:
Hi @trustin , 
Your point about differing behavior for `contains`(combining allowed) vs other options (combining not allowed , will override) is valid.
I also understand @ikhoon  point's that, when considered in isolation, `contains` can be combined (ref https://github.com/line/centraldogma/pull/809#discussion_r1115309420 ).
Personally, I am on the fence, with no strong opinion.
Perhaps @minwoox , and @jrhee17 could share their opinion :)